department-of-veterans-affairs/GIBCT-Data-Service
# Bah versioned sco

## Description

As a developer, I need to version the SCO table that is used to display SCO information in the Comparison Tool so that duplicate data is not displayed when new files are uploaded.

database migration done in a separate PR https://github.com/department-of-veterans-affairs/gibct-data-service/pull/454

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19679

front end changes https://github.com/department-of-veterans-affairs/vets-website/pull/10867

## Testing done
1. Run GIDS locally
2. Upload SCO file Multiple times
3. Check SCO table, there should only be one record of an SCO file. 
4. Check Versioned SCO table all previous generated records should be there. 
5. Publish 
6. Check Versioned SCO Archive table and all previous records should be added to this table, SCO table will have current record and Versioned SCO table will have 1 previous record. 

## Screenshots


## Acceptance criteria
- [x] When a new preview version is generated, the data is copied from "school_certifying_officials" into a versioned table.
- [x] When a preview version is published, all versions prior to the version being published are archived.


## Definition of done
